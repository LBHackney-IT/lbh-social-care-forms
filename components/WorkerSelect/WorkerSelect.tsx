import React, { Dispatch, SetStateAction } from 'react';
import s from "C:\Users\marta.pederiva_hackn\dev\lbh-social-care-frontend\components\PersonSelect\PersonSelect.module.scss';
import { format } from 'date-fns';

interface ChoiceProps {
  value: number;
  name: string;
  label: string;
  hint: string;
  idToAdd: number;
  setIdToAdd: Dispatch<SetStateAction<number>>;
}

const Choice = ({
  value,
  name,
  label,
  hint,
  idToAdd,
  setIdToAdd,
}: ChoiceProps): React.ReactElement => (
  <div className={`govuk-radios__item ${s.personResult}`}>
    <input
      className="govuk-radios__input"
      id={`${name}-${value}`}
      name={name}
      type="radio"
      value={value}
      aria-describedby={`${name}-${value}-hint`}
      checked={idToAdd === value}
      onChange={() => setIdToAdd(value)}
    />

    <label
      className={`govuk-radios__label lbh-body-s ${s.name}`}
      htmlFor={`${name}-${value}`}
    >
      {label}
    </label>

    <p id={`${name}-${value}-hint`} className={`lbh-body-xs ${s.meta}`}>
      {hint}
    </p>
  </div>
);

interface Props {
  label: string;
  people: string[];
  idToAdd: number;
  //setIdToAdd: Dispatch<SetStateAction<number>>;
}

const PersonSelect = ({
  label,
  people,
  idToAdd,
  setIdToAdd,
}: Props): React.ReactElement => {
  if (people.length === 0) return <p>No results</p>;

  return (
    <div className="govuk-form-group lbh-form-group">
      <fieldset className="govuk-fieldset">
        <legend className="govuk-label lbh-label">{label}</legend>
        <div
          className={`govuk-radios govuk-radios--small lbh-radios ${s.personList}`}
        >
          {people.map((person) => (
            <Choice
              name="person"
              label={`${person.firstName} ${person.lastName}`}
              value={
                'id' in person ? person.id : parseInt(String(person.mosaicId))
              }
              idToAdd={idToAdd}
              setIdToAdd={setIdToAdd}
              key={'id' in person ? person.id : person.mosaicId}
              hint={`#${'id' in person ? person.id : person.mosaicId} ${
                person.dateOfBirth
                  ? `· Born ${format(
                      new Date(String(person.dateOfBirth)),
                      'd MMM yyyy'
                    )} `
                  : ''
              }${
                person.address?.address ? `· ${person.address?.address}` : ''
              }`}
            />
          ))}
        </div>
      </fieldset>
    </div>
  );
};

export default PersonSelect;
