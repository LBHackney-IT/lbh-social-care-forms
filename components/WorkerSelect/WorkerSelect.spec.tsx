import WorkerSelect from './WorkerSelect';
import { render } from '@testing-library/react';

describe('workerSelect', () => {
  it('should render properly', () => {
    const mockedWorker = {
      data: {
        id: 1,
        firstName: 'Fake',
        lastName: 'Worker',
        email: 'fakeworker@foo.com',
      },
    };
    const { asFragment } = render(
      <WorkerSelect
        label="Matching workers"
        workers={[mockedWorker]}
        idToAdd={mockedWorker.id}
        setIdToAdd={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
