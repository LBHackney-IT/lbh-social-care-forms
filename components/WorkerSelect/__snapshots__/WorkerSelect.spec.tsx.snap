// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workerSelect should render properly 1`] = `
<DocumentFragment>
  <div
    class="govuk-form-group lbh-form-group"
  >
    <fieldset
      class="govuk-fieldset"
    >
      <legend
        class="govuk-label lbh-label"
      >
        Matching workers
      </legend>
      <div
        class="govuk-radios govuk-radios--small lbh-radios workerList"
      >
        <div
          class="govuk-radios__item workerResult"
        >
          <input
            aria-describedby="person-1-hint"
            class="govuk-radios__input"
            id="person-1"
            name="person"
            type="radio"
            value="1"
          />
          <label
            class="govuk-radios__label lbh-body-s name"
            for="person-1"
          >
            Fake Worker
          </label>
        </div>
      </div>
    </fieldset>
  </div>
</DocumentFragment>
`;
